udata/core/dataservices/api.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from flask import request
+from flask_login import current_user
+import mongoengine
+
+from udata.api import api, API
+from udata.api_fields import patch
+from udata.core.dataset.permissions import OwnablePermission
+from .models import Dataservice
+from udata.models import db
+
+ns = api.namespace('dataservices', 'Dataservices related operations')
+
+
+@ns.route('/', endpoint='dataservices')
+class DataservicesAPI(API):
+    @api.secure
+    @api.doc('create_dataservice', responses={400: 'Validation error'})
+    @api.expect(Dataservice.__write_fields__)
+    @api.marshal_with(Dataservice.__read_fields__, code=201)
+    def post(self):
+        dataservice = Dataservice(**request.json)
+        dataservice.owner = current_user._get_current_object()
+
+        try:
+            dataservice.save()
+        except mongoengine.errors.ValidationError as e:
+            api.abort(400, e.message)
```

Comment:
I look into the validation errors format but didn't find anything. Some endpoints use the MongoDB validation after the form validation and return the raw MongoValidation error so I thought it was ok. If the format above is the expected one from the frontend I can do the conversion.

Reply:
cc @nicolaskempf57, do you expect this payload for errors in the publishing form for example?

Reply:
I didn't encounter many nicely formatted error during udata-front development so this wasn't something we relied on most of the time because the content hasn't been really user friendly.

We only uses the API payload during a thread creation (I don't recall why but I think it can be changed to a generic error message, we're doing front-end validations to prevent form submission anyway) and during file uploads to show the error reasons to users.